### PR TITLE
Add invmod API to Integer trait

### PR DESCRIPTION
**Pull request by unseddd**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -208,6 +208,21 @@ pub trait Integer: Sized + Num + PartialOrd + Ord + Eq {
...
     }
 
+    /// Inverse of the number modulo the modulus
+    #[inline]
+    fn invmod(&self, modulus: &Self) -> Self
+    where
+        Self: Clone,
+    {
+        let egcd = self.extended_gcd(modulus);
+        assert!(egcd.gcd.is_one(), "no inverse, GCD(x, modulus) != 1");
```
Comment (cuviper) [inline src/lib.rs:218]: I think we should return `Option<Self>`, with `None` here instead of panicking.

**Review by cuviper — CHANGES_REQUESTED** [review]
This will need some unit tests, and testable doc-examples would be nice.

This also overlaps with #10 -- if @vks responds I would give preference to their PR for coming first.

File: src/lib.rs (hunk 1 above)
Comment (unseddd) [reply src/lib.rs:218]: For sure, wasn't sure what was preferred. Will change return to `Option<Self>`.

**Comment by unseddd** [comment]
> This will need some unit tests, and testable doc-examples would be nice.

No problem, can update with unit tests, and try my hand at some doc-examples :)

> This also overlaps with #10 -- if @vks responds I would give preference to their PR for coming first.

Didn't mean to step on any toes. Will definitely close, and give preference to @vks over this PR. Just wasn't sure if #10 was still relevant, since [similar code](https://github.com/rust-num/num-integer/blob/master/src/lib.rs#L165) exists in `num-integer` already (what's used by this PR).

**Comment by unseddd** [comment]
@cuviper I added the requested changes, let me know what you think.

**Comment by unseddd** [comment]
@cuviper I added the requested changes, let me know what you think. Will squash the fixup, if you approve.

**Comment by vks** [comment]
I will update my PR this week.

**Comment by unseddd** [comment]
Closing in favor of an updated #10

**Pull request closed without merging**